dotnetv3/AutoScaling/scenarios/AutoScale_Basics/AutoScale_Basics/AutoScaleBasics.cs

```diff
@@ -0,0 +1,247 @@
+System.Threading.Thread.Sleep(60000);
+
+Console.WriteLine("--- Set desired capacity to 2 ---");
+int desiredCapacity = 2;
+await AutoScaleMethods.SetDesiredCapacityAsync(client, groupName, desiredCapacity);
+
+Console.WriteLine("--- Get the two instance Id values and state ---");
+
+// Empty the group before getting the details again.
+groups.Clear();
+groups = await AutoScaleMethods.DescribeAutoScalingGroupsAsync(client, groupName);
+if (groups is not null)
+{
+    foreach (AutoScalingGroup group in groups)
+    {
+        Console.WriteLine($"The group name is {group.AutoScalingGroupName}");
+        Console.WriteLine($"The group ARN is {group.AutoScalingGroupARN}");
+        var instances = group.Instances;
+        foreach (Instance instance in instances)
+        {
+            Console.WriteLine($"The instance id is {instance.InstanceId}");
+            Console.WriteLine($"The lifecycle state is {instance.LifecycleState}");
+        }
+    }
+}
+
+Console.WriteLine("**** List the scaling activities that have occurred for the group");
+var activities = await AutoScaleMethods.DescribeAuotoScalingActivitiesAsync(client, groupName);
+if (activities is not null)
+{
+    activities.ForEach(activity =>
+    {
+        Console.WriteLine($"The activity Id is {activity.ActivityId}");
+        Console.WriteLine($"The activity details are {activity.Details}");
+    });
+}
+
+// Display the Amazon CloudWatch metrics that have been collected.
+var metrics = await CloudWatchMethods.GetCloudWatchMetricsAsync(groupName);
+Console.WriteLine($"Metrics collected for {groupName}:");
+metrics.ForEach(metric =>
+{
+    Console.Write($"Metric name: {metric.MetricName}\t");
+    Console.WriteLine($"Namespace: {metric.Namespace}");
+});
+
+var dataPoints = await CloudWatchMethods.GetMetricStatisticsAsync(groupName);
+Console.WriteLine("Details for the metrics collected:");
+dataPoints.ForEach(detail =>
+{
+    Console.WriteLine(detail);
+});
+
+// Disable metrics collection.
+Console.WriteLine("Disabling the collection of metrics for {groupName}.");
+success = await AutoScaleMethods.DisableMetricsCollectionAsync(client, groupName);
+
+if (success)
+{
+    Console.WriteLine($"Successfully stopped metrics collection for {groupName}.");
+}
+else
+{
+    Console.WriteLine($"Could not stop metrics collection for {groupName}.");
+}
+
+// Terminate all instances in the group.
+Console.WriteLine("--- Now terminating all instances in the AWS Auto Scaling group ---");
+
+groups.ForEach(group =>
+{
+    // Only delete groups in the AWS AutoScaling group we created.
+    if (group.AutoScalingGroupName == groupName)
+    {
+        group.Instances.ForEach(async instance =>
+        {
+            await AutoScaleMethods.TerminateInstanceInAutoScalingGroupAsync(client, instance.InstanceId);
+        });
+    }
+});
+
+// Once all instances have been terminated, delete the group.
```

Comment:
// After all instances are terminated, delete the group.

Reply:
changed